firefox-ios/Client/ApplicationHelper.swift

```diff
@@ -15,7 +17,33 @@ struct DefaultApplicationHelper: ApplicationHelper {
         UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!, options: [:])
     }
 
+    /// Convenience. Opens a URL with the application.
+    ///
+    /// On iPadOS if more than one window is open, the OS will
+    /// determine which UIScene the URL is delivered to.
+    /// - Parameter url: the URL to open.
     func open(_ url: URL) {
         UIApplication.shared.open(url)
     }
+
+    /// Opens a URL within a specific target window (UIScene).
+    ///
+    /// Opens a URL within a specific window on iPadOS, identified
+    /// with the provided window UUID. If no window matching the
+    /// UUID is found, the URL is opened as usual via UIApplication.open().
```

Comment:
This would be used in cases where we want to target a specific UIScene, so for ex. if you tap Show Tour and the related URL is sent via `open()` right now it potentially shows the tour on the wrong iPad window (which is part of what this PR is fixing).